### PR TITLE
JBDS-4682 - return empty array instead of default impl null

### PR DESCRIPTION
**Pull request by robstryker**
Signed-off-by: Rob Stryker <rob@oxbeef.net>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/CDKServer.java [hunk 1]
```diff
@@ -192,5 +193,11 @@ public boolean skipRegistration() {
 	public boolean skipUnregistration() {
 		return getServer().getAttribute(CDKServer.PROP_SKIP_UNREG, false);
 	}
+	
+	@Override
+	public ServerPort[] getServerPorts() {
+		return new ServerPort[0];
```
Comment (dgolovin) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/CDKServer.java]: This fixes NPE 👍 

Would that make sense to add real ports (should not be part of this PR) that could be monitored in running CDK?

Comment (robstryker) [reply plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/CDKServer.java]: I never found the WTP's port monitoring to be good at all. I don't think I was ever able to get any useful information out of it ever.

Comment (robstryker) [reply plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/CDKServer.java]: http://tools.jboss.org/blog/2015-05-28-port-monitoring.html

This is a description of how the port monitoring works. I've found in the past that it breaks a lot of ports or becomes unstable for things other than the basic scenario. 

Whats worse, though, is you're telling the tooling to contact a new random port locally when it actually wants to access otherhost:otherport.  For example, if you want to spy on your VM's EAP instance, and that instance is at 192.168.1.101:8080,   it will set up a new port on localhost (localhost:27511) and any requests sent to localhost:27511 will be forwarded over to 192.168.1.101:8080.   

So now you need to get your tooling to communicate on localhost:27511.  How would we even do that with an openshift server? Often times these servers are using the hostname from the server editor. Or they're making requests via the 'oc' command.  

Trying to get this stuff through port monitoring is a huge pain. And it's really not worth the hassle. It could necessitate a complete redesign of a lot of core assumptions.
